models/qualtrics__daily_breakdown.sql

```diff
@@ -151,17 +151,17 @@ final as (
 
     from spine
     left join agg_responses
-        on spine.date_day = agg_responses.date_day
+        on spine.date_day = agg_responses.date_day 
     left join agg_survey_distribution
-        on spine.date_day = agg_survey_distribution.date_day
+        on spine.date_day = agg_survey_distribution.date_day 
     left join agg_created_contacts
-        on spine.date_day = agg_created_contacts.date_day
+        on spine.date_day = agg_created_contacts.date_day 
     left join agg_directory_unsubscriptions
-        on spine.date_day = agg_directory_unsubscriptions.date_day
+        on spine.date_day = agg_directory_unsubscriptions.date_day 
     left join agg_mailing_list_unsubscriptions
-        on spine.date_day = agg_mailing_list_unsubscriptions.date_day
+        on spine.date_day = agg_mailing_list_unsubscriptions.date_day 
```

Comment:
hmmm yeah it's because i using the date_spine as the anchor for all the left joins, and i didn't include/add `source_relation` in the spine-creation CTE. 

i've never used this parameter before, but should we add agg by `source_relation` and then provide it as the `partition_by` argument in [date_spine](https://github.com/dbt-labs/dbt-utils#date_spine-source)?
```sql
spine as (

    {% if execute %}
    {% set first_date_query %}
        select  source_relation, min( sent_at ) as min_date from {{ var('distribution_contact') }} group by 1
    {% endset %}
    {% set first_date = run_query(first_date_query).columns[0][0]|string %}
    
    {% else %} {% set first_date = "2016-01-01" %}
    {% endif %}

    {{ dbt_utils.date_spine(
        datepart = "day", 
        start_date =  "cast('" ~ first_date[0:10] ~ "' as date)", 
        end_date = dbt.dateadd("week", 1, dbt.date_trunc('day', dbt.current_timestamp_backcompat())) ,
        partition_by="source_relation"
        ) 
    }} 
),
```

Reply:
Yeah let's give this a try. I worry with how it is built right now we may be combining data if unioned.

Reply:
ok i actually added a cross join, as the partition route was a bit more complicated and i realized we've done this in the past via a cross join

Reply:
Sounds great! Thanks for digging into this